client.go

```diff
@@ -50,9 +50,13 @@ func HttpService(rt http.RoundTripper) Service {
 		// When the calling context is cancelled, close the response body
 		// This protects callers that forget to call Close(), or those which proxy responses upstream
 		if httpRsp != nil && httpRsp.Body != nil {
-			body := httpRsp.Body
+			body := newDoneReader(httpRsp.Body)
+			httpRsp.Body = body
 			go func() {
-				<-req.Done()
+				select {
```

Comment:
I'm wondering if the tests using the leak detector could be flaky as closing happens in a separate goroutine here?

Reply:
The leak detector waits for up to a second by default for goroutines to close by default. Also, the tests definitely fail on the old version of the code (I updated the tests before writing the code to fix this issue), so I'm fairly confident it's working.